ha/lock.go

```diff
@@ -0,0 +1,132 @@
+package ha
```

Comment:
The copywrite bot is going to require a copyright header on the new files in this PR.